lib/plugins.py

```diff
@@ -27,158 +27,395 @@
+            return None                
+
+        expected_keys = {
+            'display_name': str, 'description': str, 'version': versiontuple,
+            'minimum_ec_version': versiontuple, 'package_name': str,
+        }
+        for k, expected_type in expected_keys.items():
+            v = metadata.get(k, None)
+            if v is None:
+                self.print_error("missing metadata key %s (zip plugin %s)" % (k, file_name))
+                return None                
+            if expected_type is versiontuple:
+                try:
+                    v = versiontuple(v)
+                except ValueError:
+                    self.print_error("metadata %s = %s, expected a.b.c version string (zip plugin %s)" % (k, v, file_name))
+                    return None
+            elif type(metadata[k]) is not expected_type:
+                self.print_error("metadata %s = %s, expected %s (zip plugin %s)" % (k, v, expected_type, file_name))
+                return None
+        
+        return metadata
+        
+    def install_external_plugin(self, plugin_original_path):
+        # Do the minimum verification necessary to check if the archive looks
+        # like a valid plugin zip archive.
+        metadata = self.get_metadata_from_external_plugin_zip_file(plugin_original_path)
+        if metadata is None:
+            return ExternalPluginCodes.MISSING_MANIFEST
+
+        file_name = os.path.basename(plugin_original_path)
+        leading_name, ext = os.path.splitext(file_name)
+        package_name = metadata.get("package_name", leading_name)
+        # Ensure it is not already installed.
+        if package_name in self.external_plugins or package_name in self.external_plugin_metadata:
+            return ExternalPluginCodes.NAME_ALREADY_IN_USE
+            
+        if not is_same_or_later_version(version.PACKAGE_VERSION, metadata['minimum_ec_version']):
+            return ExternalPluginCodes.INCOMPATIBLE_VERSION
+            
+        # Copy the original file to the external plugin hosting dir.
+        install_dir = self.get_external_plugin_dir()
+        plugin_file_path = os.path.join(install_dir, file_name)
+        try:
+            shutil.copyfile(plugin_original_path, plugin_file_path)
+        except OSError:
+            return ExternalPluginCodes.UNABLE_TO_COPY_FILE
+        metadata["__file__"] = plugin_file_path            
+
+        # Register the existence of the newly placed plugin archive.
+        # This would otherwise be recorded in `load_external_plugins`.
+        self.external_plugin_metadata[package_name] = metadata
+
+        # Not documented wallet type constraint.  Follow pattern elsewhere.
+        if metadata.get('requires_wallet_type'):
+            return ExternalPluginCodes.SUCCESS
+
+        # Otherwise, we enable all other installed plugins.  This causes the
+        # plugin to be loaded, afterward.
+        try:
+            self.enable_external_plugin(package_name)
```

Comment:
I almost feel like another prompt would be necessary to enable the plugin. Thats just my feeling, but usually when I install a program it doesn't start/enable immediately.

Reply:
When you install a browser plugin, it gets enabled immediately.  I think that's a better comparison than installing a program.